### ISSUE TITLE
2.2.0 release not availabel on FTP server

### ISSUE THREAD

**Issue by sebastic** [GenericMappingTools/dcw-gmt#105]
**Description of the problem**

https://www.generic-mapping-tools.org/download/ links to [DCW 2.2.0](https://github.com/GenericMappingTools/dcw-gmt/releases/download/2.2.0/dcw-gmt-2.2.0.tar.gz), which is a GitHub release artifact.

These are unfortunately unusable for the Debian package which still relies on the FTP server to find the downloads.

The release checklist includes:
>  Run make place-dcw to place the zip file on the dcw ftp and website (@pwessel).

But dcw-gmt-2.1.2.tar.gz is still the most release DCW release on the FTP server.

**Comment by joa-quim** [comment]
I don't know how we can solve this. You may not know but Paul Wessel passed away last Mars and while his ftp site is still active I doubt any of us can update files there anymore. And, as you understand, this wouldn't be a sustainable situation.

**Comment by sebastic** [comment]
I did not know that, I'm sorry for your loss.

I suspect the GitHub hosted generic-mapping-tools.org websites can't host the downloads itself, hosting copies of the downloads on the website would be an obvious choice otherwise.

We might be able to use the GitHub tags and build `dcw-gmt.nc` ourselves, but that introduces a build dependency loop as `gmt` is used by the `Makefile`.

We could also switch to monitoring https://www.generic-mapping-tools.org/download/ and live the loss of our ability to download other versions than just the latest DCW & GSHHG.

Where do you recommend packagers to look for new releases other than https://github.com/GenericMappingTools/dcw-gmt/releases?

**Comment by seisman** [comment]
> We actually have the DCW package hosted also in the GMT server (same machine that hosts [https://www.generic-mapping-tools.org](https://www.generic-mapping-tools.org/download/))

Here is the link: https://oceania.generic-mapping-tools.org/geography/dcw/

> I don't know how we can solve this.

Can we just merge the FTP server into the data server as mentioned in https://github.com/GenericMappingTools/gmtserver-admin/issues/83.

**Comment by anbj** [comment]
Out of curiosity; can you share a link to Debian’s Makefile for gmt?

**Comment by Esteban82** [comment]
The DCW is also availalbe in Zenodo. There you can get the last 3 versions:
https://zenodo.org/search?q=parent.id%3A6823056&f=allversions%3Atrue&l=list&p=1&s=10&sort=version

**Comment by sebastic** [comment]
> Can we just merge the FTP server into the data server as mentioned in [GenericMappingTools/gmtserver-admin#83](https://github.com/GenericMappingTools/gmtserver-admin/issues/83).

That sounds good, only the unpacked tree is available there currently.

> Out of curiosity; can you share a link to Debian’s Makefile for gmt?

Debian packages are more than just a Makefile, the gmt source package for Debian can be found in Salsa:

 https://salsa.debian.org/debian-gis-team/gmt/-/tree/master/debian?ref_type=heads

Which you can find via its Tracker page:

 https://tracker.debian.org/pkg/gmt

The same goes for DCW & GSHHG:

 https://tracker.debian.org/pkg/gmt-dcw
 https://tracker.debian.org/pkg/gmt-gshhg

I've switched the gmt package over to monitoring the GitHub tags like we do for many other packages, and the dcw & gshhg packages to monitoring the links on the download page as that doesn't require JavaScript unlike the GitHub releases page.

**Comment by anbj** [comment]
Thanks!

**Comment by seisman** [comment]
Not sure if it will helps the debian packaging, but at least we can upload tarballs to a server that has many mirrors.

**Comment by sebastic** [comment]
Having the historic and current tarballs on the data server will certainly help the Debian packaging. We can then import any preceding releases we may have missed.

Removing any references to the FTP server and its mirrors and only referencing the data server and its mirrors as an alternative to the downloads hosted by GitHub also simplifies things.

**Comment by joa-quim** [comment]
I agree that keeping a copy of the old (and future) source code tarballs an even the Windows installers in the data server can be useful. But **not** the full 21 GB of data that most of it useless garbage by now (old earth_relief grids, dmg installers, etc). References to ftp servers should be dropped too.
